### PR TITLE
cgen: Fix comptime checking optional type with IS operator

### LINKED ISSUE #16630: comptime: `$if` don't work
### Describe the bug

$if $else don't work for this case

### Expected Behavior

```
[./main.v:11] field: FieldData{
    name: 'a'
    attrs: []
    is_pub: false
    is_mut: false
    is_shared: false
    typ: 20
    unaliased_typ: 20
}
[./main.v:13] field: FieldData{
    name: 'b'
    attrs: []
    is_pub: false
    is_mut: false
    is_shared: false
    typ: 20
    unaliased_typ: 20
}
[./main.v:16] field: FieldData{
    name: 'b'
    attrs: []
    is_pub: false
    is_mut: false
    is_shared: false
    typ: 20
    unaliased_typ: 20
}
```

### Current Behavior

```
[./main.v:11] field: FieldData{
    name: 'a'
    attrs: []
    is_pub: false
    is_mut: false
    is_shared: false
    typ: 20
    unaliased_typ: 20
}
[./main.v:16] field: FieldData{
    name: 'b'
    attrs: []
    is_pub: false
    is_mut: false
    is_shared: false
    typ: 20
    unaliased_typ: 20
}
```

### Reproduction Steps

```v
module main

struct Foo{
	a string 
	b ?string 
}

fn main(){
	$for field in Foo.fields {
		$if field.typ is string {
			dump(field)
		} $else $if field.typ is ?string {
			dump(field) // << don't work
		}
		$if field.typ is ?string {
			dump(field) // << this work
		}
	}
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.2 1ba1f99

### Environment details (OS name and version, etc.)

ubuntu

### PR DESCRIPTION
**Pull request by felipensp**
Fix #16630

```V
module main

struct Foo{
	a string 
	b ?string
}

fn main(){
	$for field in Foo.fields {
		dump(field.name)
		$if field.typ is ?string {
			dump(1)
		} $else $if field.typ is string {
			dump(2)
		}
	}
}
```

### CODE REVIEW

File: vlib/v/checker/tests/comptime_if_optional_string_test.vv [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+fn test_main(){
+	$for field in Foo.fields {
+		dump(field.name)
+		$if field.typ is ?string {
+			dump(1)
+		} $else $if field.typ is string {
+			dump(2)
+		}
+	}
+}
```
Comment (JalonSolov) [inline vlib/v/checker/tests/comptime_if_optional_string_test.vv:17]: Looks like the fmt test needs to test .vv files, too... missing trailing EOL in this file.

Comment (felipensp) [reply vlib/v/checker/tests/comptime_if_optional_string_test.vv:17]: > Looks like the fmt test needs to test .vv files, too... missing trailing EOL in this file.

Yeah. Good catch. Thanks. 👍🏻

**Comment by JalonSolov** [comment]
Hrm... vpm site down again?

**Comment by spytheman** [comment]
> Hrm... vpm site down again?

I've rolled back the mirror to an earlier version; it should be ok now (vpm is still down).

**Pull request merged**